src/NorthwindWebsite/Core/EmailSender/EmailSender.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.AspNetCore.Identity.UI.Services;
+using Microsoft.Extensions.Options;
+using NorthwindWebsite.Core.ApplicationSettings;
+using SendGrid;
+using SendGrid.Helpers.Mail;
+
+namespace NorthwindWebsite.Core.EmailSender;
+
+public class EmailSender : IEmailSender
+{
+    private readonly AppSettings _appSettings; 
+
+    public EmailSender(
+        IOptions<AuthMessageSenderOptions> optionsAccessor,
+        AppSettings appSettings)
+    {
+        Options = optionsAccessor.Value;
+        _appSettings = appSettings;
+    }
+
+    public AuthMessageSenderOptions Options { get; } //set only via Secret Manager
+
+    public Task SendEmailAsync(string email, string subject, string message)
+    {
+        return Execute(Options.SendGridApiKey, subject, message, email);
```

Comment:
BTW, forgot to mention it earlier, you can simplify it with using lambda-expression